File: frontend/src/types/report.ts
export type ReportType = "task" | "weekly" | "final";

export interface EvaluationReport {
  id: string;
  submission_id: string | null;
  user_id: string;
  report_type: ReportType;
  architecture: Record<string, unknown> | null;
  folder_structure: Record<string, unknown> | null;
  problem_solving: Record<string, unknown> | null;
  code_quality: Record<string, unknown> | null;
  ai_usage: Record<string, unknown> | null;
  evidence: Record<string, unknown> | null;
  suggestions: string[] | null;
  summary: string | null;
  confidence: number | null;
  overall_score: number | null;
  generated_at: string;
}

export interface SkillScore {
  id: string;
  user_id: string;
  skill_name: string;
  score: number;
  source: "diagnostic" | "task_evaluation" | "viva";
  recorded_at: string;
}

export interface VivaQuestion {
  question: string;
  answer: string | null;
  followUp: string | null;
  followUpAnswer: string | null;
}

File: backend/app/schemas/report.py
from datetime import datetime

from pydantic import BaseModel


class EvaluationReportOut(BaseModel):
    """Mirrors evaluation_reports jsonb columns — the future AI review
    pipeline's response contract."""

    id: str
    submission_id: str
    architecture: dict | None = None
    folder_structure: dict | None = None
    problem_solving: dict | None = None
    code_quality: dict | None = None
    ai_usage: dict | None = None
    evidence: dict | None = None
    suggestions: list[str] = []
    confidence: float | None = None
    overall_score: float | None = None
    generated_at: datetime
